.github/workflows/cli.yml

```diff
@@ -8,13 +8,9 @@ on:
       - 'cli-*'
 
 permissions:
```

Comment:
I think its helpful to have a comment saying why these permissions are given

Reply:
Well, they aren't added in other workflows either, and also, it appears the comments are not updated with the actual settings? e.g. it says "states check reads", but it says `checks: write`.

Also, "allow repo checkout", I don't believe that needs write permission, so that too seems to be outdated? Other than that, the last two are just stating what the code already states, e.g. "Allow PRs write" vs `pull_requests: write`.

Reply:
I agree, we should update these but not remove them if they're out of date

Reply:
I can't update them because I don't know why these specific permissions are required. In my opinion, keeping outdated comments is worse than removing them (they convey information that is false).

If you have any idea what the comments should say, I'd be happy to add it back.